osu.Game/Screens/Edit/Verify/VerifyScreen.cs

```diff
@@ -118,10 +122,11 @@ protected override void LoadComplete()
 
             private void refresh()
             {
-                var issues = generalVerifier.Run(Beatmap);
+                var workingBeatmap = beatmapManager.GetWorkingBeatmap(beatmap.BeatmapInfo);
```

Comment:
You can get the `WorkingBeatmap` via DI rather than re-requesting:

```csharp
        [Resolved]
        private IBindable<WorkingBeatmap> working { get; set; }
```

Reply:
As a side note, I thought about this too, because the working beatmap instance isn't requeried by the editor when it's changed. The working beatmap cache is invalidated (see: `BeatmapManager.Save()`), but that's all.

There's a potential for discrepancies if, e.g. `BeatmapInfo` is updated on the wrong instance, because:
1. `Editor.Save()` uses the playable beatmap to save in all cases.
2. `PlayableBeatmap.BeatmapInfo` and `WorkingBeatmap.BeatmapInfo` are different instances (see: `WorkingBeatmap.GetPlayableBeatmap()`).

Just something to take note of and probably fix up - I don't like that the sections in `SetupScreen` are very careful about which beatmap to use because of this. I don't think reuse of the working beatmap causes any problems in this case...

Reply:
Yeah, definitely.

There's a hope that when switching to realm and refactoring this logic, these will actually be referencing the same instance.

Reply:
I'm not sure about that - it's done this way specifically to support mods that are `IApplicableToDifficulty`, so unless that was refactored out `BeatmapInfo`/`IBeatmap`...

Reply:
Fixed, and also double checked that editing the map and its resources properly updated all checks.

Because I use the playable beatmap in all cases apart from retrieving background texture / audio track, there shouldn't be an issues with getting an outdated `BeatmapInfo`, so reusing is probably fine. Will keep that in mind, though.